### PR TITLE
[IIIF-783] Polish festerize.py and its README

### PR DESCRIPTION
**Pull request by markmatney**
Changes:
- code cleanup
  - improve Click API usage
  - use snake for variable names instead of camel case
  - favor the new string formatting convention over the old one
- made the README a bit more concise

### CODE REVIEW

File: src/main/scripts/festerize/festerize.py [hunk 1]
```diff
@@ -10,96 +10,81 @@
...
+                click.echo(error_msg)
+                logging.error(error_msg)
                 logging.error(r.text)
                 logging.error('--------------------------------------------')
 
-        # skip any files that do no have an extension of .csv
         else:
-            click.echo("Skipping %s: not a CSV" % (fn))
-            logging.error("%s is not a CSV, skipping" % (fn))
+            error_msg= 'File {} is not a CSV, skipping'.format(csv_filename)
```
Comment (ksclarke) [inline src/main/scripts/festerize/festerize.py:86]: Is there a reason here to use the `error_msg` variable vs. putting it in the `format()` method like you did in other places?

I see on line 80 that the `click.echo()` and `logging.error()` both get the `error_msg`. Did you mean for the `logging.error()` below to get `error_msg` too (in which case, I understand the need for the variable)?

**Comment by ksclarke** [comment]
Branch is also out of date and needs to be updated against master before it can be merged.

File: src/main/scripts/festerize/festerize.py [hunk 2]
```diff
@@ -10,96 +10,81 @@
...
     if not os.path.exists(out):
-        click.echo("Output directory (%s) not found, creating it." % (out))
+        click.echo('Output directory {} not found, creating it.'.format(out))
         os.makedirs(out)
     else:
-        # alert the user, ask if they want to continue?
-        click.confirm("Output directory (%s) found, should we continue? YES might overwrite any existing output files." % (out), abort=True)
-
-    festerizer_loglevel = loglevel
+        click.confirm('Output directory {} found, should we continue? YES might overwrite any existing output files.'.format(out), abort=True)
```
[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/scripts/festerize/festerize.py:28]: I think [Pylint](https://www.pylint.org/) is the de facto standard

File: src/main/scripts/festerize/festerize.py [hunk 3]
```diff
@@ -10,96 +10,81 @@
...
-        # only work for .csv files
-        elif fn.endswith('.csv'):
+        if not csv_filepath.exists():
+            error_msg = 'File {} does not exist, skipping'.format(csv_filename)
+            click.echo(error_msg)
+            logging.error(error_msg)
 
-            click.echo("Uploading %s to %s" % (fn, request_url))
+        # Only work with CSV files that have the proper extension.
+        elif csv_filepath.suffix == '.csv':
```
[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/scripts/festerize/festerize.py:61]: I guess there's no reason other than a personal preference. I'll remove it.

File: src/main/scripts/festerize/festerize.py (hunk 1 above)
Comment (markmatney) [reply src/main/scripts/festerize/festerize.py:86]: Yes, I did. Thanks for the catch.

File: src/main/scripts/festerize/festerize.py [hunk 4]
```diff
@@ -10,96 +10,81 @@
...
+    # Logging setup.
+    started = datetime.now()
+    logfile_path = os.path.join(out, '{}.log'.format(started.strftime('%Y-%m-%d--%H-%M-%S')))
+    logging.basicConfig(filename=logfile_path, filemode='w', level=loglevel, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
 
-    # log our start time
-    logging.info("FESTERIZER STARTED at %s..." % (started.strftime("%Y-%m-%d %H:%M:%S")))
+    logging.info('STARTING at {}...'.format(started.strftime('%Y-%m-%d %H:%M:%S')))
 
-    # If Fester is unavailable, abort
```
[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/scripts/festerize/festerize.py:47]: I'll change all these name replacements back. Made them on impulse

File: src/main/scripts/festerize/README.md [hunk 5]
```diff
@@ -1,67 +1,57 @@
...
+## Installation
 
 You need to have Python 3 installed to use Festerize. This README will describe the simplest way to get started; if you're more experienced and would like to use a virtualized Python environment, feel free to do that.
 
 First, check the Fester project out of GitHub and change into the script's directory:
 
-    git clone https://github.com/UCLALibrary/fester.git
-    cd fester/src/main/scripts
+```bash
+$ git clone https://github.com/UCLALibrary/fester.git
```
[the remark this answers is not in the document]
Comment (markmatney) [reply src/main/scripts/festerize/README.md:12]: I can see that. My preference has wavered over time but I tend to like how the `$` looks. I'm happy to change back.

(In my ideal world, GitHub would apply some fancy CSS to prepend a `$` to all Bash code blocks, and that `$` would be un-highlight-able and un-copy-able by accident. Sigh ... a guy can dream.)

**Comment by markmatney** [comment]
I'll go back and apply your suggestions. All reasonable

**Comment by markmatney** [comment]
Ok, ready for re-review.

**Pull request merged**